src/solvers/refinement/string_constraint.h

```diff
@@ -0,0 +1,115 @@
+/** -*- C++ -*- *****************************************************\
+
+Module: String constraints
+        (see the PASS paper at HVC'13 and chapter 7 on arrays of ???)
+
+Author: Romain Brenguier, romain.brenguier@diffblue.com
+
+\*******************************************************************/
+
+#ifndef CPROVER_SOLVER_STRING_CONSTRAINT_H
+#define CPROVER_SOLVER_STRING_CONSTRAINT_H
+
+#include <langapi/language_ui.h>
+#include <solvers/refinement/bv_refinement.h>
+#include <solvers/refinement/refined_string_type.h>
+
+class string_constraintt : public exprt
+{
+private:
+  // String axioms can have 4 different forms:
+  // either a simple expression p,
+  // or a string constant: forall x in [0,|s|[. s(x) = c(x)
+  // or universally quantified expression: forall x in [lb,ub[. p(x)
+  // or a expression for non containment:
+  // forall x in [lb,ub[. p(x) => exists y in [lb,ub[. s1[x+y] != s2[y]
+  enum {SIMPLE, STRING_CONSTANT, UNIV_QUANT, NOT_CONTAINS} form;
+
+  // Universally quantified symbol
+  symbol_exprt quantified_variable;
```

Comment:
This has been fixed in https://github.com/diffblue/cbmc/pull/341/commits/0f6409cb740d5d2f9cab03cbd959e1b52a875a60